src/Network/Network.Test/ScenarioTests/NetworkManagerTests.ps1

```diff
@@ -698,4 +698,275 @@ function Test-NetworkManagerResourceMinimumParameterCreate
+        
+        $routingConfig = Get-AzNetworkManagerRoutingConfiguration -ResourceGroupName $rgname -NetworkManagerName $networkManagerName -Name $RoutingConfigurationName
+        Assert-NotNull $routingConfig;
+        Assert-AreEqual $RoutingConfigurationName $routingConfig.Name;
+
+        $routingConfig.Description = "A different description."
+        $routingConfig = Set-AzNetworkManagerRoutingConfiguration -InputObject $routingConfig
+        Assert-NotNull $routingConfig;
+        Assert-AreEqual "A different description." $routingConfig.Description;
+        Assert-AreEqual $RoutingConfigurationName $routingConfig.Name;
+
+        [System.Collections.Generic.List[Microsoft.Azure.Commands.Network.Models.NetworkManager.PSNetworkManagerRoutingGroupItem]]$configGroup  = @() 
+        $groupItem = New-AzNetworkManagerRoutingGroupItem -NetworkGroupId $networkGroup.Id
+        $configGroup.Add($groupItem)
+
+        New-AzNetworkManagerRoutingRuleCollection -ResourceGroupName $rgname -NetworkManagerName $networkManagerName -ConfigName $RoutingConfigurationName -Name $RuleCollectionName -AppliesTo $configGroup -DisableBgpRoutePropagation
+
+        $ruleCollection = Get-AzNetworkManagerRoutingRuleCollection -ResourceGroupName $rgname -NetworkManagerName $networkManagerName -ConfigName $RoutingConfigurationName -Name $RuleCollectionName
+        Assert-NotNull $ruleCollection;
+        Assert-AreEqual $RuleCollectionName $ruleCollection.Name;
+        Assert-AreEqual $networkGroup.Id $ruleCollection.AppliesTo[0].NetworkGroupId;
+        Assert-AreEqual "False" $ruleCollection.DisableBgpRoutePropagation;
+
+        $ruleCollection.Description = "Sample rule Collection Description"
+        $ruleCollection = Set-AzNetworkManagerRoutingRuleCollection -InputObject $ruleCollection
+        Assert-NotNull $ruleCollection;
+        Assert-AreEqual $RuleCollectionName $ruleCollection.Name;
+        Assert-AreEqual "Sample rule Collection Description" $ruleCollection.Description;
+ 
+        $destination = New-AzNetworkManagerRoutingRuleDestination -DestinationAddress "10.1.1.1/32" -Type "AddressPrefix" 
+        $nextHop = New-AzNetworkManagerRoutingRuleNextHop -NextHopType "VirtualAppliance" -NextHopAddress "2.2.2.2"
+
+        New-AzNetworkManagerRoutingRule -ResourceGroupName $rgname -NetworkManagerName $networkManagerName -ConfigName $RoutingConfigurationName -RuleCollectionName $RuleCollectionName -ResourceName $RuleName -Destination $destination -NextHop $nextHop
+
+        $routingRule = Get-AzNetworkManagerRoutingRule -ResourceGroupName $rgname -NetworkManagerName $networkManagerName -ConfigName $RoutingConfigurationName -RuleCollectionName $RuleCollectionName -Name $RuleName
+        Assert-NotNull $routingRule
+        Assert-AreEqual $RuleName $routingRule.Name 
+
+        $newRoutingRule = Set-AzNetworkManagerRoutingRule -InputObject $routingRule
```

Comment:
did you change anything about the rule here to ensure its updated? If no part of the resource is mutable, I think we should remove the Set command

Reply:
Updated the test file to modify the rule and then do a Set